### PR TITLE
test(mongo_client): add tests to show how uri parsers handle duplicate options

### PR DESCRIPTION
**Pull request by kvwalker**
Fixes NODE-1658

These are just the tests. The PR for the fix in core is [#400](https://github.com/mongodb-js/mongodb-core/pull/400)

### CODE REVIEW

File: test/functional/mongo_client_tests.js [hunk 1]
```diff
@@ -734,6 +739,109 @@ describe('MongoClient', function() {
...
+    }
+  ].forEach(test => {
+    it(test.description, function(done) {
+      const configuration = this.configuration;
+      const url = `${configuration.url()}?${test.uri}`;
+      const client = configuration.newClient(url, { useNewUrlParser: test.useNewUrlParser });
+      let stub;
+
+      if (test.useNewUrlParser) {
+        stub = sinon.stub(console, 'warn');
```
Comment (daprahamian) [inline test/functional/mongo_client_tests.js]: It would probably be a good idea to use a [`sinon.sandbox`](https://sinonjs.org/releases/v7.2.3/sandbox/) here.

Comment (kvwalker) [reply test/functional/mongo_client_tests.js]: nice, I didn't know about this!

**Comment by mbroadst** [comment]
@kvwalker can we merge this?

**Comment by kvwalker** [comment]
@mbroadst @daprahamian I just want to make sure the change to package-lock is fine with both of you and then I'll merge it!

**Comment by daprahamian** [comment]
package-lock LGTM
